src/variable.ts

```diff
@@ -67,7 +67,28 @@ export class Variable {
 	 * @private
 	 */
 	public setValue(value: number): void {
+		var previousValue = this._value
 		this._value = value
+		if (this._callback && previousValue !== value) {
+			this._callback(value, previousValue)
+		}
+	}
+
+	/**
+	 * Set a callback for whenever the value changes.
+	 *
+	 * @param {function(number,number):void} callback to call whenever the variable value changes
```

Comment:
```suggestion
	 * @param {(value: number, previousValue: number) => void} callback - The callback to call whenever the variable value changes.
```

Reply:
Ah nvm this suggestion, the jsdoc2md tool does not accept TypeScript syntax inside the type location `{}`. That's good for now.